crates/pglt_workspace/src/workspace/server/change.rs

```diff
@@ -289,7 +328,22 @@ impl Document {
             .get(usize::from(full_affected_range.start())..usize::from(full_affected_range.end()))
             .unwrap();
 
-        let new_ranges = pglt_statement_splitter::split(changed_content).ranges;
+        let (new_ranges, diags) =
+            document::split_with_diagnostics(changed_content, Some(full_affected_range.start()));
+
+        self.diagnostics = diags;
+
+        if self
+            .diagnostics
+            .iter()
+            .any(|d| d.severity() == pglt_diagnostics::Severity::Fatal)
```

Comment:
hard to see in the github diff, but is `self` not a `Document` and we can check the fatal_error helper?